packages/palette_generator/lib/palette_generator.dart

```diff
@@ -309,10 +311,10 @@ class PaletteGenerator with Diagnosticable {
     return maxScoreSwatch;
   }
 
-  PaletteColor _getMaxScoredSwatchForTarget(
+  PaletteColor? _getMaxScoredSwatchForTarget(
       PaletteTarget target, Set<Color> usedColors) {
     double maxScore = 0.0;
-    PaletteColor maxScoreSwatch;
+    PaletteColor? maxScoreSwatch;
     for (PaletteColor paletteColor in paletteColors) {
```

Comment:
@gspencergoog As an example of what I mean about propagating: is it reasonable to prevent `paletteColors` from being empty? If so, this method can be non-nullable, otherwise it can't.

Reply:
Yeah, I think it's reasonable.